conans/client/installer.py

```diff
@@ -424,8 +424,10 @@ def _build_package(self, layout, node, pref, output, keep_build, remotes):
         for python_require in conanfile.python_requires.values():
             assert python_require.ref.revision is not None, \
                 "Installer should receive python_require.ref always"
-            complete_recipe_sources(self._remote_manager, self._cache,
-                                    conanfile, python_require.ref, remotes)
+            python_conanfile = python_require.conanfile
+            python_layout = self._cache.package_layout(python_require.ref,
+                                                       short_paths=python_conanfile.short_paths)
+            complete_recipe_sources(self._remote_manager, python_layout, python_conanfile, remotes)
```

Comment:
Probably this is fixing an underlying bug: before we were using `conanfile.short_paths` to get the layout corresponding to the `python_require.ref`, now we are using the one corresponding to the `python_require`.

Much better now, but it is the symptom that we didn't have a test for this use case. 🤔